Modules/sha256module.c

```diff
@@ -503,7 +200,8 @@ SHA256Type_update(SHAobject *self, PyObject *obj)
 
     GET_BUFFER_VIEW_OR_ERROUT(obj, &buf);
 
-    sha_update(self, buf.buf, buf.len);
+    if (Hacl_Streaming_SHA2_update_256(self->state, buf.buf, buf.len) != 0)
```

Comment:
buf.len is a 64-bit Py_ssize_t, the HACL* method takes a uint32_t.  (convenient github actions compiler warning)

A loop around update seems necessary.

Reply:
Related: We don't want to raise a RuntimeError as our API allows an arbitrary Python length input.

Reply:
this also applies to the update calls in the constructors below.

Reply:
what should be the behavior if the maximum hash length is exceeded? sha2-256 is only defined for inputs of length up to 2^63-1, and undefined otherwise

Reply:
CPython isn't currently supported on any platform where `Py_ssize_t` could be larger than that $(2^{63} - 1)$ so I just wouldn't worry about it.

Have we ever had any code enforce that total sum of all update() method calls length in the past. Does anyone? It's an unreasonably long linear computation.

Reply:
Regarding handling of data > 4GB: fixed with a small helper, and a suitable test in f4b267ebe24d5b8c33001b11eaab68a8f10e575c that feeds 4GB of zeroes, followed by "hello world" into the hash function.

Regarding checking the return code: I mis-spoke, the max length is 2^61-1. But in any case, even with best-in-class accelerators, it would take over a billion years to have that much data, so I think it's ok to ignore the return value -- this is what I did in the revised version.